python/ql/src/Functions/ModificationOfParameterWithDefault.ql

```diff
@@ -93,7 +93,47 @@ class Mutation extends TaintSink {
     }
 }
 
-from TaintedPathSource src, TaintedPathSink sink
-where src.flowsTo(sink)
+/** copying prevents modification of the default value */
+class Copying extends Sanitizer {
+    Copying() { this = "Copy sanitizer"}
+
+    override predicate sanitizingDefinition(TaintKind kind, EssaDefinition def) {
+        (kind instanceof EmptyMutableValue
+        or
+        kind instanceof NonEmptyMutableValue) and
+        isCopied(def.(AssignmentDefinition).getValue())
+    }
+
+    predicate isCopied(ControlFlowNode value) {
```

Comment:
It was meant to express that `value` is something being copied. Perhaps `creates_a_copy` is a bit clearer. I would like to avoid committing to a method of copy creation :)